desktop/notification/fdo.go

```diff
@@ -0,0 +1,181 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package notification
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/godbus/dbus"
+)
+
+const (
+	dBusName          = "org.freedesktop.Notifications"
+	dBusObjectPath    = "/org/freedesktop/Notifications"
+	dBusInterfaceName = "org.freedesktop.Notifications"
+)
+
+type fdoServer struct {
+	conn *dbus.Conn
+	obj  dbus.BusObject
+}
+
+// NewFreeDesktopServer returns an interface to a freedesktop.org message notification server.
+func NewFreeDesktopServer(conn *dbus.Conn) Server {
```

Comment:
Why not just: "New()" here? Or do we plan to have non freedesktop server notifications? But given that this looks like it's super close to the FDO spec it seems a bit strange to have a "NewConsoleServer" and then having all the fde bits like Capabilities or ServerInformation etc. Or am I missing something?

Fwiw:
```
no := notifications.New()
```
reads nice I think.

Reply:
We might but I'd like to postpone this until after we've looked at the GNOME APIs.

Reply:
Aha, so there is a second NewGnomeServer() potentially? Which is a extended version of this? Or something totally different?

Reply:
GNOME is an extended version of this AFAIK